gammapy/estimators/parameter_estimator.py

```diff
@@ -0,0 +1,183 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import logging
+import numpy as np
+from gammapy.datasets import Datasets
+from gammapy.modeling import Fit
+
+__all__ = ["ParameterEstimator"]
+
+log = logging.getLogger(__name__)
+
+
+class ParameterEstimator:
+    """Model parameter estimator.
+
+    Estimates a model parameter for a group of datasets.
+    Compute best fit value, symmetric and asymmetric errors, delta TS for a given null value
+    as well as parameter upper limit and fit statistic profile.
+
+    Parameters
+    ----------
+    datasets : `~gammapy.datasets.Datasets`
+        The datasets used to estimate the model parameter
+    sigma : int
+        Sigma to use for asymmetric error computation.
+    sigma_ul : int
+        Sigma to use for upper limit computation.
+    reoptimize : bool
+        Re-optimize other free model parameters. Default is True.
+    n_scan_values : int
+        Number of values used to scan fit stat profile
+    scan_n_err : float
+        Range to scan in number of parameter error
+    """
+
+    def __init__(
+            self,
+            datasets,
+            sigma=1,
+            sigma_ul=2,
+            reoptimize=True,
+            n_scan_values=30,
+            scan_n_err=3
+    ):
+        self.sigma = sigma
+        self.sigma_ul = sigma_ul
+        self.reoptimize = reoptimize
+        self.n_scan_values = n_scan_values
+        self.scan_n_err = scan_n_err
+
+        self.datasets = self._check_datasets(datasets)
+        self.fit = Fit(datasets)
+        self.fit_result = None
+
+    def __str__(self):
+        s = f"{self.__class__.__name__}:\n"
+        s += str(self.datasets) + "\n"
+        return s
+
+    def _check_datasets(self, datasets):
+        """Check datasets geometry consistency and return Datasets object"""
+        if not isinstance(datasets, Datasets):
+            datasets = Datasets(datasets)
+
+        return datasets
+
+    def _freeze_parameters(self, datasets, parameter):
+        """Freeze all other parameters"""
+        # freeze other parameters
+        for par in datasets.parameters:
+            if par is not parameter:
+                par.frozen = True
+
+    def _compute_scan_values(self, value, value_error, par_min, par_max):
+        """Define parameter value range to be scanned"""
+        min_range = value - self.scan_n_err * value_error
+        if not np.isnan(par_min):
+            min_range = np.maximum(par_min, min_range)
+        max_range = value + self.scan_n_err * value_error
+        if not np.isnan(par_max):
+            max_range = np.minimum(par_max, max_range)
+
+        return np.linspace(min_range, max_range, self.n_scan_values)
```

Comment:
I guess for some parameters it is advantageous to set up a log-spaced grid (I think we typically do this for amplitude parameters already in the `FluxPointsEstimator`). Right now I don't have a concrete suggestion here, except that we should maybe attach the information on the "scale" to the parameter itself.